### DISCUSSION TITLE
Postgraphile v4 Integration

### DISCUSSION THREAD

**Discussion by shellscape** [dotansimha/envelop#2107] [Q&A] [answered]
Hey there. I looked far and wide, and spent a good amount of time with different search terms, but eventually came up empty and would like to ask: Does anyone have any examples or information on integrating envelop with Postgraphile v4?

I'm aware that v5 will have support for use with envelop, but we're working with v4 until v5 is in the public space.

**Comment by enisdenjo** [comment, marked as answer]
PostGraphile v4 does not create a [GraphQLSchema](https://graphql.org/graphql-js/type/#graphqlschema) which envelop (and all of its plugins) can use; it instead creates a specialised PostGraphileSchema that, in addition to being specific to PostGraphile, needs a special context to be executed within. [Read more about this in the "schema only" usage documentation.](https://www.graphile.org/postgraphile/usage-schema/)

On the other hand, PostGraphile v5 will not have this requirement and is able to create a GraphQLSchema - which in turn makes it envelop-compatible.
